backend/sozisel/lib/sozisel_web/schema/resolvers/event_resolvers.ex

```diff
@@ -0,0 +1,55 @@
+defmodule SoziselWeb.Schema.Resolvers.EventResolvers do
+  alias SoziselWeb.Context
+  alias Sozisel.Model.{Events, Sessions}
+  alias Events.Event
+  alias Sessions.Template
+
+  def create(_parent, %{input: input}, ctx) do
+    user = Context.current_user!(ctx)
+
+    with %Template{} = template <- Sessions.get_template(input.session_template_id),
+         true <- template.user_id == user.id do
+      Events.create_event(input)
+    else
+      nil ->
+        {:error, "sessions template not found"}
+
+      false ->
+        {:error, "unauthorized"}
+    end
+  end
+
+  def update(_parent, %{input: input}, ctx) do
+    user = Context.current_user!(ctx)
+
+    with %Event{} = event <- Events.get_event(input.id),
+         %Template{} = template <- Sessions.get_template(event.session_template_id),
+         true <- template.user_id == user.id do
+      input.id
+      |> Events.get_event()
+      |> Events.update_event(input)
+    else
```

Comment:
I see that you repeat that handling several times. Do something like this:
```elixir
defp handle_other(value) do
  case value do
    nil -> {:error, "sessions template not found"}
    false -> {:error, "unauthorized"}
  end
end
```
And then you can put it like this:
```elixir
with ... <- ... do
    ...sth
else
 other -> handle_other(other)
end
```
Can be `handle_error` but I don't think that values such `nil` or `false` are immediate errors but what results from them are.